include/envoy/network/connection.h

```diff
@@ -69,6 +69,37 @@ enum class ConnectionCloseType {
                      // until the delayed_close_timeout expires
 };
 
+/**
+ * Run-time modifiable connection options to alter socket settings
+ */
+class DynamicSocketOptions {
+public:
+  virtual ~DynamicSocketOptions() = default;
+
+  /**
+   * Set the receive buffer size for the socket to a value specified by the 'buffer_size'
+   */
+  virtual void setSocketRecvBufferSize(uint32_t buffer_size) const PURE;
```

Comment:
setsockopt takes ints, on linux at least.  If you pass in uint32_ts I'm concerned this will not work as expected.

Reply:
Yes. Glad that you brought this up. I was in two minds about this. I want to keep it as `int` as well. Please LMK if you dont want me to go ahead with this change.